tests/scipy_special_tests/test_lqmn.cpp

```diff
@@ -0,0 +1,152 @@
+#include "../testing_utils.h"
+
+#include <xsf/legendre.h>
+
+// `std::mdspan` (the kokkos one) was causing issues on windows and mac, so we duck-type it instead
```

Comment:
Can you give me info on what was happening here? `std::mdspan` works on all of these platforms when this code is used in SciPy. I think we should take time to figure out why this wasn't working.

Reply:
Ah that comment is outdated, gimme a sec

Reply:
Actually, I'll try it with `mdspan` again, since other things have changed now as well.

Reply:
I'm guessing that it has to do with those template `const` thingies, so that the the compiler detects the OOB indexing. 

So I guess the cleanest approach would be to also use larger `mdspan` extents, matching the buffer.